src/OrchardCore.Modules/Orchard.Tokens.Content/ContentTokens.cs

```diff
@@ -8,15 +9,15 @@ namespace Orchard.Tokens
 {
     public static class ContentTokens
     {
-        public static void RegisterContentTokens(this IHandlebars handlebars)
+        public static void RegisterContentTokens(this IHandlebars handlebars, IHttpContextAccessor httpContextAccessor)
         {
             // Renders a slug for the current content item. Do not use to represent the url
             // as the content item as it might be different than the computed slug.
             handlebars.RegisterHelper("slug", (output, context, arguments) =>
             {
-                IServiceProvider serviceProvider = context.ServiceProvider;
-                var contentManager = serviceProvider.GetRequiredService<IContentManager>();
-                var slugService = serviceProvider.GetRequiredService<ISlugService>();
+                var services = httpContextAccessor.HttpContext.RequestServices;
```

Comment:
No they don't. It was kind of a hack to pass the ServiceProvider to the context, and I prefer the new solution for two reasons:
- We don't have to mix ViewData and services in the context object.
- The context might be an object which is not extensible (not dynamic) and it would not work.

The next issue is that if we need to execute these tokens in a non web context like a background task we'll need to provide an IHttpContextAccessor. But I assume we would have to do it anyway, like in O1. But now I can see that we'd have an issue as the current IHttpContextAccessor is store only once on these templates. Or we should use a different Handlebars instances for background tasks.